test/com/lynxanalytics/biggraph/frontend_operations/AggregateOnNeighborsTest.scala

```diff
@@ -51,3 +51,24 @@ class AggregateOnNeighborsTest extends OperationsTestBase {
       2 -> Vector(41, -74, 48, 19)))
   }
 }
+
+class WeightedAggregateOnNeighborsTest extends OperationsTestBase {
+  test("all aggregators") {
+    def agg[T](attribute: String, aggregator: String, weight: String): Map[Long, T] = {
+      val p = box("Create example graph")
+        .box(
+          "Weighted aggregate on neighbors",
+          Map(
+            "prefix" -> "",
+            "direction" -> "all edges",
+            "weight" -> weight,
+            ("aggregate_" + attribute) -> aggregator))
+        .project
+      get(p.vertexAttributes(attribute + "_" + aggregator + "_by_" + weight)).asInstanceOf[Map[Long, T]]
```

Comment:
I think this should also work:
```suggestion
      get(p.vertexAttributes(attribute + "_" + aggregator + "_by_" + weight).runtimeSafeCast[T])
```